### PR TITLE
Fix a few issues with some rules we noticed when migrating to flat config

### PR DESCRIPTION
**Pull request by christiango**
I work on an internal repo at Microsoft that uses these rules. We noticed that when upgrading to eslint flat config that some eslint errors were no longer firing. After some debugging we found that the following issues issues:

* When type of an object was Window | null, no-postmessage-star-origin did not report it as an error
* When the type of an HTML element was just "Element" no-inner-html didn't fire.

### CODE REVIEW

File: lib/rules/no-inner-html.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ module.exports = {
 
     function mightBeHTMLElement(node) {
       const type = astUtils.getNodeTypeAsString(fullTypeChecker, node, context);
-      return type.match(/HTML.*Element/) || type === "any";
+      return type.match(/HTML.*Element/) || type === "any" || type == "Element";
```
Comment (plade) [inline lib/rules/no-inner-html.js]: Could we avoid treating every TypeScript type rendered as `Element` as the DOM type? A local `class Element { innerHTML = ""; }` also renders as `Element`, so this addition reports a non-DOM assignment. A narrow fix would be to keep the existing heuristic but, for this exact case, verify that the type symbol has a declaration in `lib.dom.d.ts`. Please also add a valid regression test for a local `Element` and an invalid test for the DOM `Element` type.

Comment (christiango) [reply lib/rules/no-inner-html.js]: [Generated by GPT-5.6 sol]

Updated the exact `Element` case to inspect its TypeScript symbol declarations and only treat it as the DOM type when a declaration comes from `lib.dom.d.ts`. This prevents a local `class Element` from being reported. I added a valid regression test for the local type and an invalid regression test for the DOM type.

I also enabled `strictNullChecks` in the TypeScript test fixture and added explicit `Window | null` coverage so the union-handling change is exercised rather than having TypeScript collapse the union to `Window`. The full suite passes with 142 tests.

File: lib/rules/no-postmessage-star-origin.js [hunk 2]
```diff
@@ -9,6 +9,10 @@
 
 const astUtils = require("../ast-utils");
 
+function isWindowOrAny(type) {
+  return type === "any" || type === "Window";
```
Comment (plade) [inline lib/rules/no-postmessage-star-origin.js]: Same issue as earlier applies. Window can be a custom defined class, and need to be verified against the DOM.

Comment (christiango) [reply lib/rules/no-postmessage-star-origin.js]: [AI-generated response by GPT-5.6 Sol]

Addressed in `eba455e`. The rule now verifies that an exact `Window` type has a symbol declaration in `lib.dom.d.ts` rather than relying only on TypeScript’s rendered type name. This check is also applied to each union/intersection constituent, so DOM `Window | null` remains covered while a locally defined `class Window` is ignored.

I extracted the declaration-origin check into `ast-utils` and reused it for the existing `Element` handling. I also added a valid regression test for a custom `Window`; it failed before the implementation and passes afterward. The full suite passes with 143 tests.

**Review by plade — APPROVED** [review]
